### ISSUE TITLE
PullRequestDetailState is null when switching changed file list view

### ISSUE THREAD

**Issue by paladique** [github/VisualStudio#710]
Testing `grokys/maintainer-workflow-ui-update` 

When changing the view of files (options button) in the detail view, I get a null reference exception in the `PullRequestDetailState` at this [line](https://github.com/github/VisualStudio/blob/grokys/maintainer-workflow-ui-update/src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs#L124)

The view does end up changing (list or tree) when I continue debugging.

**Comment by grokys** [comment]
Just pushed a new commit to attempt to fix that. Could you check if it works?

**Comment by paladique** [comment]
Just gave this a try, still getting the exception

**Comment by paladique** [comment]
Found the problem for my particular case. I had to delete the registry key settings for the `UIState` before the fix in that last commit worked for me. Since the state was initially set to null when I built and ran master earlier this week, it's always null in my settings.

**Comment by grokys** [comment]
Registry key settings? Not sure what you mean here - what exactly did you have to delete?

**Comment by paladique** [comment]
Wrong wording, apologies, it was very late 😅  There's a registry key called UIState that I needed to delete. It's value contains JSON with other UIState classes (GitHubConnectSectionState, RepositoryUIState) and also the key/value pair `"PullRequestDetailState":null`, which never gets updated because the settings can't be saved when PullRequestDetailState is null. 

Once I deleted the key and ran the branch with the fix, the UIState key was created again and "PullRequestDetailState" had its default values.
